compiler/compiler/src/run.rs

```diff
@@ -70,43 +76,176 @@ pub struct Case {
+            program_name: case.program_name.clone(),
+            function: case.function.clone(),
+            status: EvaluationStatus::Failed(e),
+            output: Value::make_unit(),
+        };
+
+        let vm = VM::from(ConsensusStore::<CurrentNetwork, ConsensusMemory<CurrentNetwork>>::open(
+            StorageMode::Production,
+        )?)?;
+
+        vm.process().write().add_programs_with_editions(&programs_and_editions)?;
+
+        let private_key = PrivateKey::from_str(leo_ast::TEST_PRIVATE_KEY).unwrap();
+        let program_id = ProgramID::<CurrentNetwork>::from_str(&case.program_name).unwrap();
+        let function_id = Identifier::<CurrentNetwork>::from_str(&case.function).unwrap();
+        let inputs = case.input.iter();
+
+        // --- catch panics from authorize ---
+        let authorization_result =
+            catch_unwind(AssertUnwindSafe(|| vm.authorize(&private_key, program_id, function_id, inputs, rng)));
+
+        let authorization = match authorization_result {
+            Ok(auth_res) => match auth_res {
+                Ok(auth) => auth,
+                Err(e) => {
+                    outcomes.push(bad_outcome(format!("{e}")));
+                    continue; // go to the next case 
+                }
+            },
+            Err(e) => {
+                outcomes.push(bad_outcome(format!("{e:?}")));
+                continue; // go to the next case 
+            }
+        };
+
+        // --- catch panics from evaluate ---
+        let eval_result =
+            catch_unwind(AssertUnwindSafe(|| vm.process().read().evaluate::<AleoTestnetV0>(authorization)));
+
+        let response = match eval_result {
+            Ok(eval_res) => match eval_res {
+                Ok(resp) => resp,
+                Err(e) => {
+                    outcomes.push(bad_outcome(format!("{e}")));
+                    continue; // go to the next case 
+                }
+            },
+            Err(e) => {
+                outcomes.push(bad_outcome(format!("{e:?}")));
+                continue; // go to the next case 
+            }
+        };
+
+        let outputs = response.outputs();
+        let output = match outputs.len() {
+            0 => Value::make_unit(),
+            1 => outputs[0].clone().into(),
+            _ => Value::make_tuple(outputs.iter().map(|x| x.clone().into())),
+        };
+
+        outcomes.push(Outcome::EvaluationOutcome {
+            program_name: case.program_name.clone(),
+            function: case.function.clone(),
+            status: EvaluationStatus::Success,
+            output,
+        });
+    }
+
+    Ok(outcomes)
 }
 
 /// Run the functions indicated by `cases` from the programs in `config`.
 // Currently this is used both by the test runner in `test_execution.rs`
 // as well as the Leo test in `cli/commands/test.rs`.
 // `leo-compiler` is not necessarily the perfect place for it, but
 // it's the easiest place for now to make it accessible to both of those.
```

Comment:
Looks like this might apply to a lot of items in this module. Could this be refactored out into some shared mod for testing? Otherwise I can imagine users browsing the pub APIs in docs.rs, trying to use `run_*` fns not realising they're just for tests. Maybe best as a follow-up issue.